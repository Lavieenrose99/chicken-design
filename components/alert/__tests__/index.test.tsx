import React from 'react';
import { render } from '@testing-library/react';
import Alert from '../alert';

describe('<Alert />', () => {
  test('should render default', () => {
    const { container } = render(<Alert>default</Alert>);
    expect(container).toMatchSnapshot();
  });

  test('should render alert with type', () => {
    const kinds: any[] = ['info', 'warning', 'positive', 'negative'];

    const { getByText } = render(
      <>
        {kinds.map(k => (
          <Alert kind={k} key={k}>
            {k}
          </Alert>
        ))}
      </>,
    );

    kinds.forEach(k => {
      expect(getByText(k)).toMatchSnapshot();
    });
  });
});
