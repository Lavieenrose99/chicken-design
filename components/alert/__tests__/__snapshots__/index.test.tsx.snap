// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> should render alert with type 1`] = `
<div
  class="happy-alert"
  style="background: rgb(83, 82, 237);"
>
  info
</div>
`;

exports[`<Alert /> should render alert with type 2`] = `
<div
  class="happy-alert"
  style="background: rgb(255, 165, 2);"
>
  warning
</div>
`;

exports[`<Alert /> should render alert with type 3`] = `
<div
  class="happy-alert"
  style="background: rgb(46, 213, 115);"
>
  positive
</div>
`;

exports[`<Alert /> should render alert with type 4`] = `
<div
  class="happy-alert"
  style="background: rgb(255, 71, 87);"
>
  negative
</div>
`;

exports[`<Alert /> should render default 1`] = `
<div>
  <div
    class="happy-alert"
    style="background: rgb(83, 82, 237);"
  >
    default
  </div>
</div>
`;
